app/services/feature_service.rb

```diff
@@ -1,26 +1,10 @@
-module FeatureService
-  class FormRequiredError < StandardError; end
+class FeatureService
+  class << self
+    def enabled?(feature_name)
+      return false if Settings.features.blank?
 
-  def self.enabled?(feature_name, form = nil)
-    return false if Settings.features.blank?
-
-    segments = feature_name.to_s.split(".")
-    feature = Settings.features.dig(*segments)
-
-    return feature unless feature.is_a? Config::Options
-
-    return true if feature.enabled
-
-    if feature.enabled_for_form_ids.present?
-      raise FormRequiredError, "Feature #{feature_name} requires form to be provided" if form.blank?
-
-      return feature.enabled_for_form_ids == form.id if feature.enabled_for_form_ids.is_a?(Integer)
-
-      form_ids = feature.enabled_for_form_ids.split(",").collect(&:strip)
-
-      return form_ids.include?(form.id.to_s)
+      segments = feature_name.to_s.split(".")
+      segments.reduce(Settings.features) { |config, segment| config[segment] }
```

Comment:
Let's use `dig` instead of `reduce` here:

```suggestion
      Settings.features.dig(*segments)
```